packages/legend-query-builder/src/stores/QueryBuilderState.ts

```diff
@@ -444,12 +461,42 @@ export abstract class QueryBuilderState implements CommandRegistrar {
     return PrimitiveType.STRING;
   }
 
-  initializeWithQuery(query: RawLambda): void {
-    this.rebuildWithQuery(query);
+  initializeWithQuery(
+    query: RawLambda,
+    defaultParameterValues?: Map<string, ValueSpecification>,
+  ): void {
+    this.rebuildWithQuery(query, {
+      defaultParameterValues,
+    });
     this.resetQueryResult();
     this.changeDetectionState.initialize(query);
   }
 
+  buildInitialParameterValues(
```

Comment:
please inline this function, it's not being used any where else, and the file is already super-cluttered.